### PR TITLE
Add card masking to visual experiment

### PR DESCRIPTION
**Pull request by bbaker6225**
[REV-911](https://openedx.atlassian.net/browse/REV-911)

Supported cards come from: https://support.edx.org/hc/en-us/articles/206502498-How-do-I-pay-
Card formats taken from http://wysiwyg.co.il/Anatomy-of-CreditCard-Number-formats.asp and just searching around

**Visa Format**
Screenshot_2019-10-31 Payment edX

**AMEX Format**:
Screenshot_2019-10-31 Payment edX(1)

### CODE REVIEW

File: src/payment/checkout/payment-form/CardDetails.jsx [hunk 1]
```diff
@@ -22,6 +23,24 @@ export class CardDetailsComponent extends React.Component {
     };
   }
 
+  componentDidMount() {
+    if (this.props.isPaymentVisualExperiment) {
+      const cardFormats = [
+        '4999 9{4} 9{4} 9{4}', // Visa
```
Comment (bbaker6225) [inline src/payment/checkout/payment-form/CardDetails.jsx]: For `inputmask`, `9` represents any number and `{4}` represents 4 of the same thing. So this line corresponds to:
`4xxx xxxx xxxx xxxx` where `x` is any number

**Comment by bbaker6225** [comment]
@MatthewPiatetsky Interesting point. That's probably more a question for legal, since I believe similar PCI concerns were what led to the card validator package being moved inside this repo

File: package-lock.json [hunk 2]
```diff
@@ -7916,14 +7915,12 @@
         "balanced-match": {
           "version": "1.0.0",
           "bundled": true,
-          "dev": true,
-          "optional": true
```
Comment (julianajlk) [inline package-lock.json]: Not sure what update this refers to but should this be in the commit?

**Comment by bbaker6225** [comment]
I have reached out to legal to see if there are any PCI concerns with this

File: package-lock.json (hunk 2 above)
Comment (bbaker6225) [reply package-lock.json]: I know it's an unsatisfying answer, but this is just what I get from running `npm install inputmask`. 🤷‍♂

**Comment by bbaker6225** [comment]
We are going to hold off on adding this to the experiment for now, pending an in-depth PCI review of the package

**Pull request closed without merging**